src/gretel_trainer/benchmark/gretel/models.py

```diff
@@ -83,3 +85,90 @@ class GretelGPTX(GretelModel):
 
 class GretelDGAN(GretelModel):
     config = "synthetics/time-series"
+
+
+class _GretelModelWithOverrides(GretelModel):
+
+    _delegate: GretelModel
+    _name: str
+    _extra_config: Optional[dict]
+
+    def __init__(self,
+                 delegate: GretelModel,
+                 name: str = "",
+                 extra_config: Optional[dict] = None,
+    ):
+        self._delegate = delegate
+        self._name = name
+        self._extra_config = extra_config
+
+    @property
+    def name(self) -> str:
+        return self._name if self._name else self._delegate.name
+
+    @property
+    def config(self) -> GretelModelConfig:
+        if not self._extra_config:
+            return self._delegate.config
+
+        cfg = copy.deepcopy(read_model_config(self._delegate.config))
+        model_cfg = next(c for c in cfg["models"][0].values())
+        _recursive_dict_update(model_cfg, self._extra_config)
+        return cfg
+
+    @property
+    def model_key(self) -> str:
+        return self._delegate.model_key
+
+    @property
+    def trainer_model_type(self) -> Optional[gretel_trainer.models._BaseConfig]:
+        return self._delegate.trainer_model_type
+
+    def runnable(self, dataset: Dataset) -> bool:
+        return self._delegate.runnable(dataset)
+
+
+def configure_model(
+        model: Union[Type[GretelModel], GretelModel],
+        name: str = "",
+        extra_config: Optional[dict] = None,
```

Comment:
nit: "extra_" suggests to me that it can only be additive, rather than override existing/default keys in the config. Would "override_" or "update_" be a better prefix?

Reply:
Point taken, but I wanted to avoid `override_` because to me it suggests the opposite :) Maybe `update_` is a good compromise

Reply:
Changed to `config_update`